go/vt/vttablet/tabletmanager/rpc_vreplication_test.go

```diff
@@ -568,3 +572,173 @@ func TestUpdateVReplicationWorkflow(t *testing.T) {
 		})
 	}
 }
+
+// TestFailedMoveTablesCreateCleanup tests that the workflow
+// and its artifacts are cleaned up when the workflow creation
+// fails -- specifically after the point where we have created
+// the workflow streams.
+func TestFailedMoveTablesCreateCleanup(t *testing.T) {
+	ctx := context.Background()
+	sourceKs := "sourceks"
+	sourceTabletUID := 200
+	shard := "0"
+	targetTabletUID := 300
+	targetKs := "targetks"
+	wf := "testwf"
+	table := defaultSchema.TableDefinitions[0].Name
+	invalidTimeZone := "NOPE"
+	bls := fmt.Sprintf("keyspace:\"%s\" shard:\"%s\" filter:{rules:{match:\"%s\" filter:\"select * from %s\"}}",
+		sourceKs, shard, table, table)
+	tenv := newTestEnv(t, sourceKs, []string{shard})
+	defer tenv.close()
+	ws := workflow.NewServer(tenv.ts, tenv.tmc)
+
+	sourceTablet := tenv.addTablet(t, sourceTabletUID, sourceKs, shard)
+	defer tenv.deleteTablet(sourceTablet.tablet)
+	targetTablet := tenv.addTablet(t, targetTabletUID, targetKs, shard)
+
```

Comment:
we need a defer to delete the target tablet as well?

Reply:
Yeah, good catch! I'll add that. ❤️